lib/rules/selector-not-notation/index.mjs

```diff
@@ -78,48 +78,49 @@ const rule = (primary, _, context) => {
 
 			if (!isStandardSyntaxSelector(selector)) return;
 
-			const fixedSelector = parseSelector(selector, result, ruleNode, (container) => {
-				container.walkPseudos((pseudo) => {
-					if (!isNot(pseudo)) return;
-
-					if (primary === 'complex') {
-						const prev = pseudo.prev();
-						const hasConsecutiveNot = prev && isNot(prev);
-
-						if (!hasConsecutiveNot) return;
-
-						if (context.fix) return fixComplex(prev);
-					} else {
-						const selectors = pseudo.nodes;
-
-						if (isSimple(selectors)) return;
-
-						const mustFix =
-							context.fix &&
-							selectors.length > 1 &&
-							selectors[1] &&
-							(selectors[1].nodes.length === 0 ||
-								selectors.every(({ nodes }) => nodes.length === 1));
-
-						if (mustFix) return fixSimple(pseudo);
-					}
-
-					assert(pseudo.source && pseudo.source.end);
-
-					report({
-						message: messages.expected,
-						messageArgs: [primary],
-						node: ruleNode,
-						index: pseudo.sourceIndex,
-						endIndex: pseudo.source.end.column,
-						result,
-						ruleName,
-					});
+			const selectorRoot = parseSelector(selector, result, ruleNode);
+
+			if (!selectorRoot) return;
+
+			selectorRoot.walkPseudos((pseudo) => {
+				if (!isNot(pseudo)) return;
+
+				if (primary === 'complex') {
+					const prev = pseudo.prev();
+					const hasConsecutiveNot = prev && isNot(prev);
+
+					if (!hasConsecutiveNot) return;
+
+					if (context.fix) return fixComplex(prev);
+				} else {
+					const selectors = pseudo.nodes;
+
+					if (isSimple(selectors)) return;
+
+					const mustFix =
+						context.fix &&
+						selectors.length > 1 &&
+						selectors[1] &&
+						(selectors[1].nodes.length === 0 || selectors.every(({ nodes }) => nodes.length === 1));
+
+					if (mustFix) return fixSimple(pseudo);
+				}
+
+				assert(pseudo.source && pseudo.source.end);
+
+				report({
+					message: messages.expected,
+					messageArgs: [primary],
+					node: ruleNode,
+					index: pseudo.sourceIndex,
+					endIndex: pseudo.source.end.column,
+					result,
+					ruleName,
 				});
 			});
 
-			if (context.fix && fixedSelector) {
```

Comment:
note

This was a potential bug.

https://github.com/stylelint/stylelint/blob/713f36f1f7e02723e1949524e53aab7b000853c4/lib/rules/selector-not-notation/__tests__/index.mjs#L35-L36

could potentially return `''` if it ever was supported.